integration/util_test.go

```diff
@@ -60,3 +61,26 @@ func copyFile(src, dst string) error {
 	}
 	return w.Sync()
 }
+
+type conditionFunc func() (bool, error)
+
+// poll calls a condition function repeatedly on a polling interval until it returns true, returns an error
+// or the timeout is reached. If the condition function returns true or an error before the timeout, poll
+// immediately returns with the true value or the error. If the timeout is exceeded, poll returns false.
+func poll(interval time.Duration, timeout time.Duration, condition conditionFunc) (bool, error) {
```

Comment:
can you move this to pkg/testutil/testutil.go?

Reply:
Ah, yes, that's what I need.